clang/test/CodeGenSYCL/sub-group-size.cpp

```diff
@@ -15,15 +15,15 @@ using namespace sycl;
 // ALL-DAG: define {{.*}}spir_func void @{{.*}}external_10{{.*}}() #{{[0-9]+}}{{.*}} !intel_reqd_sub_group_size ![[TEN:[0-9]+]]
 
 SYCL_EXTERNAL void external_default_behavior() {}
-// NONE-DAG: define {{.*}}spir_func void @{{.*}}external_default_behavior{{.*}}() #{{[0-9]+}} !srcloc !{{[0-9]+}} !sycl_fixed_targets !{{[0-9]+}} {
```

Comment:
@steffenlarsen It was necessary to remove these `sycl_fixed_targets` attrs after changing the flag to get this to pass. This feels like a regression and breaks the test due to [this change](https://github.com/intel/llvm/pull/7701/files#diff-680aac42e651b506634e9cf972770658b7a5119e233a168c11ff7c8edc443e56) moving the first SYCL aspects pass into the `DisableLLVMPasses` if and putting it behind the `!CodeGenOpts.DisableSYCLEarlyOpts` check. Does this need to change?

Reply:
I believe the aspect propagation pass should be exempt from `-fno-sycl-early-optimizations` similar to how it was exempt from `-disable-llvm-passes` before.

Reply:
Agree, `PropagateAspectsUsage` is a functional pass, it should still be enabled even if the flag is present